### PR TITLE
gendata.py with torchvision.datasets.MNIST

### PR DESCRIPTION
**Pull request by pochoi**
`tensorflow.examples.tutorials.mnist` is deprecated. I have replaced it with `torchvision.datasets.MNIST`.

### CODE REVIEW

**Comment by antigol** [comment]
Thanks! Did you check that it works the same ? Is the normalization of the data the same ? (Just to be sure)

**Comment by pochoi** [comment]
I merely replace `tensorflow.examples.tutorials.mnist` by `torchvision.datasets.MNIST` and nothing else is done. It is not normalized, as in the original example, i.e. image values on range 0 to 255.

I tested it in Google Colaboratory, since it provides convenient GPU access.
[https://drive.google.com/file/d/1O3zXW63VjEefYVAui81bnfHF0YL1iNga/view?usp=sharing](https://drive.google.com/file/d/1O3zXW63VjEefYVAui81bnfHF0YL1iNga/view?usp=sharing)
I run it on the unrotated images. CNN gets 97% accuracy and Spherical CNN gets 96%. It seems working.

By the way, I actually did think about the normalization on the images, but it is more sophisticated and requires more testing, so I think it should be a separate commit. Here is my primary thought. Average on S2 is indeed the (l=0, m=0) spherical harmonics coefficient, up to the normalization convention on spherical harmonics. Therefore we already have all the ingredient to do average. Then we can normalize any spherical image in this way:
```python
from s2cnn.soft.s2_fft import S2_fft_real
import torch

def s2_average(x):
  return S2_fft_real(1)(x)[0,0]

x = torch.rand(60,60)*255
x_mean = s2_average(x)
x_sd = s2_average((x - x_mean).pow(2)).sqrt()
y = (x - x_mean)/x_sd

s2_average(y) # should be 0.0
s2_average(y.pow(2)) # should be 1.0
```
Do you think normalizating spherical images in this way makes sense?

**Comment by antigol** [comment]
Yes it makes sense
